spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala

```diff
@@ -73,6 +73,44 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
    */
   var gradInput: A = Activity.allocate[A, T]()
 
+
+  private[bigdl] var inputShapeValue: Activity = null
+
+  private[bigdl] var outputShapeValue: Activity = null
+
+  /**
+   * There's no batch dim in the inputShape which just represent a sample record.
+   */
+  def getInputShape(): Activity = {
+    inputShapeValue
+  }
+
+  /**
+   * There's no batch dim in the outputShape which just represent a sample record.
+   */
+  def getOutputShape(): Activity = {
+    if (outputShapeValue == null) {
+      throw new RuntimeException("You should build this model first before getting the outputshape")
+    }
+    outputShapeValue
+  }
+
+  def isBuilt(): Boolean = {
```

Comment:
why we need this?

Reply:
The idea is to tell if the given layer is a "concrete" layer or not. but it's useless for now. Would remove it.

Reply:
what do you mean "concrete" layer...